hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -100,9 +99,11 @@ protected BaseLoadBalancer(MetricsBalancer metricsBalancer) {
     this.metricsBalancer = (metricsBalancer != null) ? metricsBalancer : new MetricsBalancer();
   }
 
-  @Override
-  public void setConf(Configuration conf) {
-    this.config = conf;
+  protected final Configuration getConf() {
+    return provider.getConfiguration();
+  }
+
+  protected void setConf(Configuration conf) {
```

Comment:
I wonder after `setconf`, maybe `getconf` is expired result, so `setconf` should be none parameter, only use provider's conf ?

Reply:
There is another call path, from ConfigurationObserver.onConfigurationChange, so here we need to keep this method as is.

Filed HBASE-25852 for moving the intialization work all to initialize method, which could make the implementation more clear.